src/lake_modelling/utils/ca_model.py

```diff
@@ -0,0 +1,210 @@
+    res = np.array([dClake_dt])
+
+    return res
+
+
+def run_ca_model(
+    init_cond,
+    lake,
+    lim_param,
+    products,
+    dt=0.01,
+):
+    """Run the model for Ca concentration using parameters from 'par_dict'.
+    Args
+        par_dict: Dict of model parameters:
+                      par_dict = {
+                            "C_lake0": C_lake0,
+                            "C_in0": C_in0,
+                            "C_bott0": C_bott0,
+                            "pH_lake0": pH_lake0,
+                            "lime_prod": lime_prod,
+                            "lime_dose": lime_dose,
+                            "lime_month": lime_month,
+                            "spr_meth": spr_meth,
+                            "K_L": K_L,
+                            "F_active": F_active,
+                            "area": area,
+                            "mean_depth": mean_depth,
+                            "res_time": res_time,
+                            "flow_profile": flow_profile,
+                            "n_months": n_months,
+                        }
+        dt:         Float between 0 and 1 (months). Time resolution in decimal
+                    months for evaluating the model within each monthly time step.
+                    NOTE: This parameter does not affect how the ODEs are solved
+                    (that is handled automatically). It simply sets the level of
+                    temporal detail in the output. Larger values run faster, but
+                    give coarser output. Default 0.01.
+    Returns
+        Dataframe with time index (in decimal months) and Ca concentration.
+    """
+    # Unpack parameters
+    C_lake0, C_in0, C_bott0, pH_lake0 = init_cond
+    area, mean_depth, res_time, flow_profile = lake
+    lime_prod, lime_dose, lime_month, spr_meth, K_L, F_active, n_months = lim_param
+    par_dict = {
+        "C_lake0": C_lake0,
+        "pH_lake0": pH_lake0,
+        "lime_prod": lime_prod,
+        "lime_dose": lime_dose,
+        "spr_meth": spr_meth,
+        "K_L": K_L,
+        "F_active": F_active,
+        "res_time": res_time,
+        "flow_profile": flow_profile,
+    }
+
+    LIME_DATA = products
+
+    if spr_meth == "dry":
+        spr_fac = LIME_DATA[lime_prod]["DryFac"]
+    else:
+        spr_fac = 1
+
+    # Estimate instantaneous dissolution
+    col_depth = LIME_DATA[lime_prod]["ColDepth"]
+    depth_corr = np.log10(mean_depth / col_depth)
+    ph_col_ca = pH_lake0 - depth_corr
+    od_fac = LIME_DATA[lime_prod][find_nearest("OD", lime_dose)]
+    inst_diss_pct = (
+        spr_fac * LIME_DATA[lime_prod][find_nearest("ID", ph_col_ca)] / od_fac
+    )
+    ca_dose = lime_dose * LIME_DATA[lime_prod]["CaPct"] / 100
+    C_inst = ca_dose * inst_diss_pct / 100
+
+    # Hydrology
+    V = 1000 * area * 1e6 * mean_depth  # Lake volume in litres
+    q_mean = V / (12 * res_time)  # Annual mean flow in litres/month to give res_time
```

Comment:
Related to the `res_time` input in `user_inputs.py`. Shall we somehow make sure that this value is never 0 so it doesn't result in error (division by 0)? Maybe setting the minimum value in the `user_inputs.py` to 0.1?

Reply:
done